x/concentrated-liquidity/lp.go

```diff
@@ -89,15 +92,22 @@ func (k Keeper) CreatePosition(ctx sdk.Context, poolId uint64, owner sdk.AccAddr
 		return CreatePositionData{}, err
 	}
 
-	positionId := k.getNextPositionIdAndIncrement(ctx)
-
```

Comment:
Note: this was moved below the `hasPositions` check to ensure the before hook was triggered after validation logic but before any state changes (`getNextPositionIdAndIncrement` mutates state)